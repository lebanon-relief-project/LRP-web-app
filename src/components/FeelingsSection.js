import React from "react";
import styled from "styled-components";
import colours from "../styles/Colours";

const OurMission = () => {
  return (
    <StyledSection>
      <Wrapper>
        <StyledHeader>Learn more and select the ones that you have experienced</StyledHeader>
        <StyledParagraph>
            Below are flashcards with the most common presentations of traumatic stress.
            Click on those that you have experienced to learn more about your reaction and ways to
            manage.
        </StyledParagraph>
      </Wrapper>
    </StyledSection>
  );
};

const Wrapper = styled.div`
  width: 50%;
  padding: 5rem 1rem;
  margin: 0 12rem;
  background: inherit;
`;

const StyledSection = styled.section`
  background: ${colours.white};
  margin: 0;
  padding: 0;
  width: 100%;
`;

const StyledHeader = styled.h2`
  font-weight: bold;
  font-size: 20px;
`;

const StyledParagraph = styled.p`
  margin: 0;
  font-size: 14px;
  
`;

export default OurMission;
