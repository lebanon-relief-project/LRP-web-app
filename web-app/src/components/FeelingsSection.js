import React from "react";
import styled from "styled-components";
import colours from "../styles/Colours";
import devices from "../styles/Devices";

const FeelingsSection = () => {
  return (
    <StyledSection>
      <Wrapper>
        <StyledHeader>Something feels different</StyledHeader>
        <TextWrapper>
          <StyledParagraph>
            If you feel that you are experiencing emotional and physical changes
            since a distressing event, know that you are not alone, and that it
            is normal to be wanting to understand your changes after a traumatic
            event. The cards below give you a brief description of what you may
            be experiencing. Please read them and select those that apply to
            you. Once you are done, and want to know more about why you are
            experiencing those feelings and how to cope with them, click on give
            me advice at the bottom of the page.
          </StyledParagraph>
        </TextWrapper>
      </Wrapper>
    </StyledSection>
  );
};

const TextWrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
`;

const Wrapper = styled.div`
  width: 60%;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: 0 20%;
  background: inherit;

  @media (max-width: ${devices.ipadpro}) {
    width: auto;
    width: 75%;
    margin: 0 12.5%
  }

  @media (max-width: ${devices.ipad}) {
    margin: 0 30px;
    width: 90%;
  }

  @media (max-width: ${devices.mobile}) {
    margin: 25px 18px;
    padding-top: 0px;
    padding-bottom: 0px;
    width: auto;
  }
`;

const StyledSection = styled.section`
  background: ${colours.white};
  margin: 0;
  padding: 0;
  width: 100%;
`;

const StyledHeader = styled.h2`
  font-weight: bold;
  font-size: 24px;
`;

const StyledParagraph = styled.p`
  margin: 0;
  font-size: 16px;
`;

export default FeelingsSection;
