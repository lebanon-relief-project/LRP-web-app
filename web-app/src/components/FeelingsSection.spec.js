import React from "react";
import { render } from "@testing-library/react";
import FeelingsSection from "./FeelingsSection";
import "jest-styled-components";

describe("the FeelingsSection component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderFeelingsSection());

    expect(container).toMatchSnapshot();
  });
});

const renderFeelingsSection = () => {
  return render(<FeelingsSection />);
};
