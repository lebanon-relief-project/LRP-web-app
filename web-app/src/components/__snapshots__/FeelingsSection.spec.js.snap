// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FeelingsSection component should match the snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  width: 100%;
  padding-top: 2rem;
  padding-bottom: 2rem;
  margin: auto;
  background: inherit;
  max-width: 960px;
}

.c0 {
  background: #ffffff;
  margin: 0;
  padding: 0;
  width: 100%;
}

.c2 {
  font-weight: bold;
  font-size: 24px;
}

.c4 {
  margin: 0;
  font-size: 16px;
}

@media (max-width:1025px) {
  .c1 {
    width: auto;
    width: 75%;
    margin: 0 12.5%;
  }
}

@media (max-width:768px) {
  .c1 {
    margin: 0 30px;
    width: 90%;
  }
}

@media (max-width:576px) {
  .c1 {
    margin: 25px 18px;
    padding-top: 0px;
    padding-bottom: 0px;
    width: auto;
  }
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Something feels different
      </h2>
      <div
        class="c3"
      >
        <p
          class="c4"
        >
          If you feel that you are experiencing emotional and physical changes since a distressing event, know that you are not alone, and that it is normal to be wanting to understand your changes after a traumatic event. The cards below give you a brief description of what you may be experiencing. Please read them and select those that apply to you. Once you are done, and want to know more about why you are experiencing those feelings and how to cope with them, click on give me advice at the bottom of the page.
        </p>
      </div>
    </div>
  </section>
</div>
`;
